File: java/src/test/java/fr/soat/cleancoders/exo5/GameTest.java
package fr.soat.cleancoders.exo5;

import org.assertj.core.api.Assertions;
import org.junit.Test;

public class GameTest {

    @Test
    public void testDefaultMove() {
        Game game = new Game("XOXOX-OXO");

        int result = game.move('X');

        Assertions.assertThat(result).isEqualTo(5);

        game = new Game("XOXOXOOX-");
        result = game.move('O');

        Assertions.assertThat(result).isEqualTo(8);

        game = new Game("---------");
        result = game.move('X');

        Assertions.assertThat(result).isEqualTo(0);

        game = new Game("XXXXXXXXX");
        result = game.move('X');

        Assertions.assertThat(result).isEqualTo(-1);
    }


    @Test
    public void testFindWinningMove() {
        Game game = new Game("XO-XX-OOX");

        int result = game.move('X');

        Assertions.assertThat(result).isEqualTo(5);
    }

    @Test
    public void testWinConditions() {
        Game game = new Game("---XXX---");

        char winner = game.winner();

        Assertions.assertThat(winner).isEqualTo('X');
    }
}

File: csharp/Exo4/GameTest.cs
using NFluent;
using Xunit;

namespace Soat.CleanCoders.Refactoring.Exo4
{
    public class GameTest
    {
        [Fact]
        public void TestDefaultMove()
        {
            var game = new Game("XOXOX-OXO");

            var result = game.Move('X');

            Check.That(result)
                 .IsEqualTo(5);

            game   = new Game("XOXOXOOX-");
            result = game.Move('O');

            Check.That(result)
                 .IsEqualTo(8);

            game   = new Game("---------");
            result = game.Move('X');

            Check.That(result)
                 .IsEqualTo(0);

            game   = new Game("XXXXXXXXX");
            result = game.Move('X');

            Check.That(result)
                 .IsEqualTo(-1);
        }

        [Fact]
        public void TestFindWinningMove()
        {
            var game = new Game("XO-XX-OOX");

            var result = game.Move('X');

            Check.That(result)
                 .IsEqualTo(5);
        }

        [Fact]
        public void TestWinConditions()
        {
            var game = new Game("---XXX---");

            var winner = game.Winner();

            Check.That(winner)
                 .IsEqualTo('X');
        }
    }
}
